### ISSUE TITLE
Reading mode is not working properly

### ISSUE THREAD

**Issue by charles-prof** [mdn/yari#7648]
### MDN URL

https://developer.mozilla.org/en-US/docs/MDN/Community/Issues

### What specific section or headline is this issue about?

reading mode does not work properly

### What information was incorrect, unhelpful, or incomplete?

Web assembly introduction page is not showing up properly in [reading mode](https://developer.mozilla.org/en-US/docs/WebAssembly).

https://developer.mozilla.org/en-US/docs/WebAssembly

![image](https://user-images.githubusercontent.com/68528137/203459745-31b9e0ca-f4a2-4d33-ba77-c2fc59abfaee.png)

### What did you expect to see?

properly formatted text (webpage) for reading

### Do you have any supporting links, references, or citations?

no

### Do you have anything more you want to share?

This is a simple issue. reporting for first time is overwhelming. 

Additional question forum link or community chat link would be helpful to easily ask what this (what I see or experience in this web app) means

because of the doubt whether it is a firefox bug to be reported in bugzilla or here.

even a mail support is good for asking

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/mdn/community/issues`
* MDN URL: https://developer.mozilla.org/en-US/docs/MDN/Community/Issues
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/mdn/community/issues/index.md
* Last commit: https://github.com/mdn/content/commit/6d83d1a87aa6480f4ef5aae29ee50cfe6a8d47a2
* Document last modified: 2022-11-10T15:17:18.000Z

</details>

**Comment by caugner** [comment]
Hi @charles-prof, thank you for taking the time to report this issue. 👋 

This is similar to https://github.com/mdn/yari/issues/5358, but much worse, because the most important content sections of [the page](https://developer.mozilla.org/en-US/docs/WebAssembly) (namely the introduction, among others) are omitted in the Firefox Reader View.

As mentioned in [this comment](https://github.com/mdn/yari/issues/5358#issuecomment-1315510658):

> The Firefox team pointed me to https://github.com/mozilla/readability/blob/master/Readability.js, which is the underlying implementation that decides what elements to show.

What may be happening here is that the high number of `<dl>`/`<dt>`/`<dd>` elements causes the Firefox Reader View to give that section's ancestors such a low content score (see [here](https://github.com/mozilla/readability/blob/fde359445cc96888fd8fa9717d336558ace83a5d/Readability.js#L764-L773)) that the comparatively few text in the sections above don't increase it enough (see [here](https://github.com/mozilla/readability/blob/fde359445cc96888fd8fa9717d336558ace83a5d/Readability.js#L1028-L1029)), and so other nodes are picked instead.

I'm not sure there is much that we can do, but it's definitely worth opening an issue in https://github.com/mozilla/readability/, referencing this issue. 🙏

**Comment by caugner** [comment]
FWIW Safari Reader View does not have this issue:

<img width="818" alt="image" src="https://user-images.githubusercontent.com/495429/203857418-7cbf0179-f015-4448-bb03-7fbcbb3c0378.png">

**Comment by caugner** [comment]
Chrome Reader Mode (behind a flag) is not affected either:

<img width="696" alt="image" src="https://user-images.githubusercontent.com/495429/203857766-d3cc4c1b-852a-4887-b1e9-501202e2396b.png">

**Comment by charles-prof** [comment]
> I'm not sure there is much that we can do, but it's definitely worth opening an issue in https://github.com/mozilla/readability/, referencing this issue. 🙏

Thanks for pointing me in the right direction. 

As I was trying to inspect the console before reporting. The page rendered correctly in reader mode.

I checked the console. ( i turned off vue.js devtools to check if extension is making this but thst's not the case)

![image](https://user-images.githubusercontent.com/68528137/204569514-933a01ee-7578-46bd-ac0a-afcba9d17489.png)

Layout is forced before page loading. - This is making the content unreadable but if i hit f5 to refresh the page, the content loads correctly.

This only affects the initial loading when we hit f9 or entering reading mode using icon in address bar.

**Comment by caugner** [comment]
> This only affects the initial loading when we hit f9 or entering reading mode using icon in address bar.

Good catch, I have now filed this as [bug 1803192](https://bugzilla.mozilla.org/show_bug.cgi?id=1803192) in the Firefox bug tracker.

**Comment by caugner** [comment]
FWIW The issue [has been reported](https://github.com/mozilla/readability/issues/774) in the Firefox Readability repository in December 2022.

**Comment by janbrasna** [comment]
This no longer reproduces for me, using FXDE 138.0beta on macOS15:

> <img width="927" alt="Image" src="https://github.com/user-attachments/assets/dbcc54fa-1082-4cb1-9696-ef6103d6df55" />
